src/Microsoft.AspNetCore.Authentication/AuthenticationOptions.cs

```diff
@@ -47,5 +47,13 @@ public string AuthenticationScheme
         /// Additional information about the authentication type which is made available to the application.
         /// </summary>
         public AuthenticationDescription Description { get; set; } = new AuthenticationDescription();
+
+        /// <summary>
+        /// Defines whether access and refresh tokens should be stored in the
+        /// <see cref="AuthenticationProperties"/> after a successful authorization.
+        /// This property is set to <c>false</c> by default to reduce
+        /// the size of the final authentication cookie.
+        /// </summary>
+        public bool SaveTokens { get; set; }
```

Comment:
So the only one that doesn't use this now is Cookies...  And there's nothing preventing that from saving tokens either...

Its basically a flag that tells the auth middleware to save tokens into AuthenticationProperties if it happens to have any tokens.  If JwtBearer can use this flag, I don't see why its not generically applicable.  AuthenticationMiddleware who don't have any tokens can just ignore this option...